megamek/src/megamek/client/bot/princess/Princess.java

```diff
@@ -1863,7 +1862,11 @@ CardinalEdge getHomeEdge(Entity entity) {
         }
         
         // otherwise, return the destination edge
-        return getBehaviorSettings().getDestinationEdge();
+        if (getBehaviorSettings().getDestinationEdge() == CardinalEdge.NEAREST) {
+            return BoardUtilities.getClosestEdge(entity);                
+        } else {
+            return getBehaviorSettings().getDestinationEdge();
+        }
```

Comment:
Ternary candidate

Reply:
I tried it and it doesn't look that much more readable in ternary form.